flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java

```diff
@@ -397,14 +397,14 @@ public void close() {
     }
 
     private void startKafkaContainerCluster(int numBrokers) {
-        Network network = Network.newNetwork();
         if (numBrokers > 1) {
+            Network network = Network.newNetwork();
```

Comment:
I think a cleaner way to avoid 'ERROR: could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network' is release the created network resources after used it.

Reply:
Thanks for the review @leonardBang ! Fixed in e4e1d24.